components/brave_federated/task/model_util.cc

```diff
@@ -0,0 +1,29 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated/task/model_util.h"
+
+#include <cmath>
+
+namespace brave_federated {
+
+float ComputeNLL(std::vector<float> true_labels,
+                 std::vector<float> predictions) {
+  float error = 0.0;
+  size_t batch_size = true_labels.size();
+
+  for (size_t i = 0; i < batch_size; i++) {
+    error += (true_labels[i] * log(predictions[i]) +
```

Comment:
What if true_labels is a different size than batch_size?

Reply:
did you mean `predictions` size? Because we define batch_size based on `true_labels` size. Added a CHECK_EQ on the dimensionality of the two input vectors.

Reply:
Any index that could be different based on external meta data or across different functions, would need these checks. Best to add defensive code for such things that are downloaded etc., i.e. out of our control and CHECK for those situations where should never happen